### PR TITLE
add rate limiting handler

### PR DESCRIPTION
**Pull request by pnikonowicz**
This handler is a rabbitMQ only solution for rate limiting. It can decorate other handlers (ie: OneShot or MaxRetry)
to allow rate limiting as well as error handling.

Look at the ratelimiter.md for further explanation

Signed-off-by: Paul Nikonowicz <pnikonowicz@pivotal.io>

### CODE REVIEW

File: lib/sneakers/handlers/maxretry.rb [hunk 1]
```diff
@@ -87,6 +87,8 @@ def acknowledge(hdr, props, msg)
       end
 
       def reject(hdr, props, msg, requeue = false)
+        Sneakers.logger.info{"In maxretry reject. #{props.inspect}"}
```
Comment (michaelklishin) [inline lib/sneakers/handlers/maxretry.rb]: Please change the severity of these to `debug`

**Comment by pnikonowicz** [comment]
> This likely conflicts with some ideas in #364 🤔

I might be able to use the middleware, or extend it if needed in order to get the hook I need.

**Comment by pnikonowicz** [comment]
I'm not sure why the CI check is failing. It appears to be failing on other PR's as well. Is this failure valid?

**Comment by pnikonowicz** [comment]
Master branch is failing tests. Assume that when master branch passes, this branch will pass as well.

**Comment by pnikonowicz** [comment]
conflicts resolved
build passing

**Pull request closed without merging**